app/models/concerns/recordable_deletion.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# Module to add to models that should record their own deletion in the DeletedResource collection.
+# The identifer defaults to the id, but can be specified via the identifies_deleted_resources_by
+# @example using Mongo uuids to track deleted resources
+#   class Dummy
+#     include RecordableDeletion
+#     include Mongoid::Uuid
+#
+#     identifies_deleted_resource_by :uuid
+#
+#
+module RecordableDeletion
+  extend ActiveSupport::Concern
+
+  included do
+    before_destroy :create_deleted_resource
```

Comment:
This concern at present creates DeletedResources for:
* any web resource
* only contributions never having been published

What it needs to do is create DeletedResources for:
* only web resources of contributions ever published
* only contributions ever published (supplementing the retained Contribution record with the deleted_at and deleted_by)

We have no use case for retaining data about contributions and web resources never published.

Perhaps we should remove this callback from the concern and make it the responsibility of each including class to define when it should run, with the necessary conditional logic for its data model, in particular given that for the Contribution model this would need to go into an AASM callback, rather than an ActiveModel one.